internal/pkg/agent/application/upgrade/upgrade.go

```diff
@@ -626,3 +633,17 @@ func IsInProgress(c client.Client, watcherPIDsFetcher func() ([]int, error)) (bo
 
 	return state.State == cproto.State_UPGRADING, nil
 }
+
+// isSameReleaseVersion will if upgradeVersion and currentVersion are equal using only release numbers.
+// They are not equal if either are a SNAPSHOT, or if the semver numbers (including prerelease and build identifiers) differ.
+func isSameReleaseVersion(current agentVersion, upgradeVersion string) bool {
+	if current.snapshot {
+		return false
+	}
+	target, _ := agtversion.ParseVersion(upgradeVersion)
```

Comment:
This will be the first time that the upgradeVersion is parsed, please handle the error.
`IsSameVersion` had an explicit comment about being able to skip the error of parsing the version string again because it was already parsed and handled in lines 198-201 but here we have no such guarantee as this is called on line 172